src/as_execution/abi.rs

```diff
@@ -20,11 +20,20 @@ macro_rules! get_memory {
     ($env:ident) => {
         match $env.get_ffi_env().memory.as_ref() {
             Some(mem) => mem,
-            _ => abi_bail!("No memory in env"),
+            _ => abi_bail!("Environment memory is missing"),
         }
     };
 }
 
+pub(crate) fn get_env(ctx: &FunctionEnvMut<ASEnv>) -> ABIResult<ASEnv> {
+    let env = ctx.data().clone();
+    if *env.available.write() == false {
+        abi_bail!("ABI calls are not available during instantiation");
+    } else {
```

Comment:
Can we add a function here to factorize?

Reply:
I'm not sure what you mean by that